### ISSUE TITLE
Invalid code inside `object` block compiles

### ISSUE THREAD

**Issue by mgwidmann** [absinthe-graphql/absinthe#399]
I'm not sure whats going on here. I noticed that the following module compiles perfectly fine and I'm kind of confused as to why... I've cut out the non-relevant parts but you can see the whole file in the link at the bottom.

```elixir
defmodule SlackCoder.GraphQL.Schemas.RandomFailure do
  use Absinthe.Schema.Notation
  use Absinthe.Ecto, repo: SlackCoder.Repo

  object :failure do
    @desc "The primary identifier for each failure."
    field :id, :id

    # ... removed ...

    As.long(as: this.is(valid: :syntax).compile.errors, will: not complain)
  end
end
```

You can see invalid code here (accidentally copied from my ecto schema):

https://github.com/mgwidmann/slack_coder/blob/ec28dd4c598105614acd97bf24ac4dfecebceba6/lib/graphql/schemas/random_failure_object.ex#L51

This compiles fine 😕  ... Any ideas how thats happening?

**Comment by benwilson512** [comment]
Hey!

This is unfortunately a known limitation of the way in which the schema macros work at the moment. The schema macros basically walk the AST within them and expand any other macros, but nothing other than macros is ever actually run or even returned to the module body. Consequently, it swallows code like this.

I'm gonna close this because there's really nothing we can do about it until 1.5 when we do our schema rework, and it is a known limitation.

**Comment by mgwidmann** [comment]
I see... Good to know it'll be fixed in 1.5 otherwise I'd recommend to throw exceptions for common things like my example.
